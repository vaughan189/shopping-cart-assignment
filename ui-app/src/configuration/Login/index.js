import {
  emailRegexPattern,
  passwordRegexPattern,
} from "../../constants/validation";

export const loginConfiguration = [
  {
    id: "email-field",
    label:"Email",
    name: "email-field",
    type: "text",
    validation: {
      name: "email",
      required: true,
      pattern: emailRegexPattern,
    },
    error: {
      id: "email-error-text",
      name: "email",
      message: "Please enter a valid email",
    },
  },
  {
    id: "password-field",
    label:"Password",
    name: "password-field",
    type: "password",
    validation: {
      name: "password",
      required: true,
      minLength: 6,
      pattern: passwordRegexPattern,
    },
    error: {
      id: "password-error-text",
      name: "password",
      message: "Please enter a valid password",
    },
  },
];
