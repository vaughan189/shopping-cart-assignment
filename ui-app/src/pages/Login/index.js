import React from "react";
import {
  Button,
  Container,
  FormControl,
  FormHelperText,
  Grid,
  Input,
  InputLabel,
  Typography,
} from "@mui/material";

import { useForm } from "react-hook-form";
import {
  emailRegexPattern,
  passwordRegexPattern,
} from "../../constants/validation";
import { useHistory } from "react-router-dom";

export const Login = () => {
  let history = useHistory();
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm();

  const onSubmit = () => {
    history.push("/home");
  };

  return (
    <Container maxWidth="xl">
      <Grid container direction="row" spacing={10} sx={{ marginTop: "0%" }}>
        <Grid item xs="auto" sm="auto" md={2} xl={2}></Grid>
        <Grid item xs={12} sm={12} md={4} xl={4}>
          <Grid
            container
            direction="row"
            justifyContent="flex-start"
            alignItems="flex-start"
          >
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <Typography
                sx={{
                  fontFamily: "Dosis",
                  fontSize: 28,
                  fontWeight: 600,
                  justifyContent: "centre",
                }}
              >
                Login
              </Typography>
            </Grid>
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <Typography
                sx={{
                  fontFamily: "Dosis",
                  fontSize: 18,
                  fontWeight: 400,
                }}
              >
                Get access to your Orders. Wishlist and Recommendation
              </Typography>
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12} sm={12} md={4} xl={4}>
          <Grid
            container
            direction="row"
            justifyContent="flex-start"
            alignItems="flex-start"
          >
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <FormControl
                variant="standard"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                }}
              >
                <InputLabel htmlFor="email">Email</InputLabel>
                <Input
                  id="email"
                  type="text"
                  {...register("email", {
                    required: true,
                    pattern: emailRegexPattern,
                  })}
                />
                {errors.email && (
                  <FormHelperText
                    id="component-error-text"
                    sx={{ color: "red" }}
                  >
                    Please enter a valid email
                  </FormHelperText>
                )}
              </FormControl>
            </Grid>
            <Grid item xs={12} sm={12} md={12} xl={12}>
              <FormControl
                variant="standard"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                }}
              >
                <InputLabel htmlFor="password">Password</InputLabel>
                <Input
                  id="password"
                  type="password"
                  {...register("password", {
                    required: true,
                    minLength: 6,
                    pattern: passwordRegexPattern,
                  })}
                />
                {errors.password && (
                  <FormHelperText
                    id="component-error-text"
                    sx={{ color: "red" }}
                  >
                    Please enter a valid password
                  </FormHelperText>
                )}
              </FormControl>
            </Grid>
            <Grid item xs={12} md={12}>
              <Button
                variant="contained"
                sx={{
                  width: { xl: "100%", md: "100%", sm: "90%", xs: "90%" },
                  margin: 2,
                  fontFamily: "Dosis",
                  textTransform: "none",
                }}
                onClick={handleSubmit(onSubmit)}
                disabled={errors.email || errors.password}
              >
                Login
              </Button>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Container>
  );
};
