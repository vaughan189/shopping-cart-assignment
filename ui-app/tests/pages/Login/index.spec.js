import React from "react";
import reactRouterDom from "react-router-dom";
import { shallow } from "enzyme";
import Login from "../../../src/pages/Login";

jest.mock("react-router-dom");
const pushMock = jest.fn();
reactRouterDom.useHistory = jest.fn().mockReturnValue({ push: pushMock });

describe("Login Page", () => {
  it("should render the Login Page correctly", () => {
    const component = shallow(<Login debug />);
    expect(component).toMatchSnapshot();
  });

  it("should trigger the submit function on button click", () => {
    const component = shallow(<Login debug />);
    component.find("#email-field").simulate("change", {
      target: { value: "test@example.com", name: "email-field" },
    });
    component.find("#password-field").simulate("change", {
      target: { value: "Pass@123456", name: "password-field" },
    });

    component.find("#login-btn").simulate("click");
    // expect(pushMock).toHaveBeenCalled();
  });

  it("should trigger an error message if a wrong emailID is entered", () => {
    const component = shallow(<Login debug />);
    component.find("#email-field").simulate("change", {
      target: { value: "test@example", name: "email-field" },
    });
    component.find("#password-field").simulate("change", {
      target: { value: "Pass@123456", name: "password-field" },
    });
    component.find("#login-btn").simulate("click");
    // expect(component.find("#email-error-text").text()).toBe(
    //   "Please enter a valid email"
    // );
  });
});
