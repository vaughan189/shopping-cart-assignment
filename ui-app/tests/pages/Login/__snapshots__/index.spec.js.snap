// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render the Login Page correctly 1`] = `
<ForwardRef(Container)
  maxWidth="xl"
>
  <ForwardRef(Grid)
    container={true}
    direction="row"
    spacing={10}
    sx={
      Object {
        "marginTop": "0%",
      }
    }
  >
    <ForwardRef(Grid)
      item={true}
      md={2}
      sm="auto"
      xl={2}
      xs="auto"
    />
    <ForwardRef(Grid)
      item={true}
      md={4}
      sm={12}
      xl={4}
      xs={12}
    >
      <ForwardRef(Grid)
        alignItems="flex-start"
        container={true}
        direction="row"
        justifyContent="flex-start"
      >
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(Typography)
            sx={
              Object {
                "fontFamily": "Dosis",
                "fontSize": 28,
                "fontWeight": 600,
                "justifyContent": "centre",
              }
            }
          >
            Login
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(Typography)
            sx={
              Object {
                "fontFamily": "Dosis",
                "fontSize": 18,
                "fontWeight": 400,
              }
            }
          >
            Get access to your Orders. Wishlist and Recommendation
          </ForwardRef(Typography)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      md={4}
      sm={12}
      xl={4}
      xs={12}
    >
      <ForwardRef(Grid)
        alignItems="flex-start"
        container={true}
        direction="row"
        justifyContent="flex-start"
      >
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(FormControl)
            sx={
              Object {
                "margin": 2,
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="standard"
          >
            <ForwardRef(InputLabel)
              htmlFor="email-field"
            >
              Email
            </ForwardRef(InputLabel)>
            <ForwardRef(Input)
              id="email-field"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </ForwardRef(FormControl)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          sm={12}
          xl={12}
          xs={12}
        >
          <ForwardRef(FormControl)
            sx={
              Object {
                "margin": 2,
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="standard"
          >
            <ForwardRef(InputLabel)
              htmlFor="password-field"
            >
              Password
            </ForwardRef(InputLabel)>
            <ForwardRef(Input)
              id="password-field"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              type="password"
            />
          </ForwardRef(FormControl)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          md={12}
          xs={12}
        >
          <ForwardRef(Button)
            id="login-btn"
            onClick={[Function]}
            sx={
              Object {
                "fontFamily": "Dosis",
                "margin": 2,
                "textTransform": "none",
                "width": Object {
                  "md": "100%",
                  "sm": "90%",
                  "xl": "100%",
                  "xs": "90%",
                },
              }
            }
            variant="contained"
          >
            Login
          </ForwardRef(Button)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</ForwardRef(Container)>
`;
